src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectionRuleOptionsTests.cs

```diff
@@ -1517,6 +1517,73 @@ public async Task CollectionRuleOptions_GetEnvironmentVariable_NamePropertyValid
                 });
         }
 
+        [Fact]
+        public async Task CollectionRuleOptions_CollectStacksAction_FeatureDisabled()
+        {
+            await ValidateFailure(
+                rootOptions =>
+                {
+                    const string fileEgress = nameof(fileEgress);
+                    rootOptions.AddFileSystemEgress(fileEgress, "/tmp")
+                        .CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddCollectStacksAction(fileEgress);
+                },
+                ex =>
+                {
+                    string[] failures = ex.Failures.ToArray();
+                    Assert.Single(failures);
+                    VerifyFeatureDisabled(failures, 0);
+                });
+        }
+
+        [Fact]
+        public async Task CollectionRuleOptions_CollectStacksAction_FeatureDisabledByFlag()
+        {
+            await ValidateFailure(
+                rootOptions =>
+                {
+                    const string fileEgress = nameof(fileEgress);
+                    rootOptions.AddFileSystemEgress(fileEgress, "/tmp")
+                        .EnableInProcessFeatures()
+                        .CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddCollectStacksAction(fileEgress);
+                },
+                ex =>
+                {
+                    string[] failures = ex.Failures.ToArray();
+                    Assert.Single(failures);
+                    VerifyFeatureDisabled(failures, 0);
+                });
+        }
+
+        [Fact]
+        public async Task CollectionRuleOptions_CollectStacksAction_FeatureEnabled()
+        {
+            try
+            {
+                Environment.SetEnvironmentVariable(ExperimentalFlags.Feature_CallStacks, "true");
```

Comment:
I wonder if we need to refactor the experimental feature flag to be a service that is replaceable by testing so that we don't have to do this (not expecting you to do it in this change).

Reply:
Oddly enough, this doesn't seem to work in the CI, but does work on my local test run.

Reply:
It looks like since the feature flag's state is a lazy static; once a test evaluates the flag to be false, another test in the same hosting process will also think it's false.